test/dynamo/test_higher_order_ops.py

```diff
@@ -3729,6 +3732,106 @@ def test_non_aliasing_util(self):
         ):
             _assert_tensors_nonaliasing(a, a)
 
+    @torch._dynamo.config.patch(
+        for_loop_medium_size_boundary=10
```

Comment:
@Fidget-Spinner I rewrote this to be something we can scope to the test, to make for more manageable/easier testing. Lmk if you hate it.